tests/integration/component.go

```diff
@@ -205,6 +205,29 @@ func componentTests(args ...string) {
 				project, "--binary", filepath.Join(context, "sb.jar"))
 		})
 
+		It("binary component should fail when --binary is not in --context folder", func() {
+			oc.ImportJavaIS(project)
+			helper.CopyExample(filepath.Join("binary", "java", "openjdk"), context)
+
+			newContext := helper.CreateNewContext()
+			defer helper.DeleteDir(newContext)
```

Comment:
@kadel Does this line will be executed if in case line ```Expect(output).To(ContainSubstring("inside of the context directory"))``` fails ? If yes then we are fine, otherwise a context dir will be left behind each time.

Reply:
Yes, defer is executed when the function where this is called ends. Even if tests end with en error when this `It` block ends the defer function will be called.